internal/http/services/owncloud/ocdav/prop/prop.go

```diff
@@ -58,13 +59,78 @@ func EscapedNS(namespace string, local string, val string) PropertyXML {
 	}
 }
 
-// Escaped returns a new PropertyXML instance while xml-escaping the value
+var (
+	escAmp  = []byte("&amp;")
+	escLT   = []byte("&lt;")
+	escGT   = []byte("&gt;")
+	escFFFD = []byte("\uFFFD") // Unicode replacement character
+)
+
+// Decide whether the given rune is in the XML Character Range, per
+// the Char production of https://www.xml.com/axml/testaxml.htm,
+// Section 2.2 Characters.
+func isInCharacterRange(r rune) (inrange bool) {
+	return r == 0x09 ||
+		r == 0x0A ||
+		r == 0x0D ||
+		r >= 0x20 && r <= 0xD7FF ||
+		r >= 0xE000 && r <= 0xFFFD ||
+		r >= 0x10000 && r <= 0x10FFFF
+}
+
+// Escaped returns a new PropertyXML instance while replacing only
+// * `&` with `&amp;`
+// * `<` with `&lt;`
+// * `>` with `&gt;`
+// as defined in https://www.w3.org/TR/REC-xml/#syntax:
+//
+// > The ampersand character (&) and the left angle bracket (<) must not appear
+// > in their literal form, except when used as markup delimiters, or within a
+// > comment, a processing instruction, or a CDATA section. If they are needed
+// > elsewhere, they must be escaped using either numeric character references
+// > or the strings " &amp; " and " &lt; " respectively. The right angle
+// > bracket (>) may be represented using the string " &gt; ", and must, for
+// > compatibility, be escaped using either " &gt; " or a character reference
+// > when it appears in the string " ]]> " in content, when that string is not
+// > marking the end of a CDATA section.
+//
+// The code ignores errors as the legacy Escaped() does
 // TODO properly use the space
 func Escaped(key, val string) PropertyXML {
```

Comment:
I would suggest to move this to a dedicated pkg. We want the search REPORT endpoint to be consistent with the PROPFIND endpoint. But we can't import it from `internal` path

Reply:
moving some of the ocdav specific stuff to `pkg` makes sense, right. extracting a dedicated oc flavoured webdav package as `github.com/owncloud/go-webdav` that others can reuse is the right way to go.

But not as part of this PR.